### PR TITLE
Quote v2: update how attributes are registered

### PR DESCRIPTION
**Pull request by oandregal**
Follow-up to #39703 and #39704 

## What?

This PR updates how the new attribute for Quote v2 is registered. As a side-effect, the v2 quote block declares all the block supports the v1 has and the `align` attribute as well.

## Why?

Using a separate `block.json` for v2 is problematic because it's not picked up automatically.

## How?

Instead of having a separate `block.json` for v2, this uses the `blocks.registerBlockType` hook to register the new attribute `attribution` and unregister the old one `citation`.

## Testing Instructions

Check that attributes are registered properly is v2 is active:

- Go to "Gutenberg > Experiments" page and enable the v2 of the quote block.
- Load the editor.
- Verify that the `core/blocks` store contains the proper block attributes under `blockTypes.core/quote`:
  -  it should not have the `value` and `citation` attributes
  - it should have the `attribution` attribute

Check that attributes are registered properly if v1 is active:

- Go to "Gutenberg > Experiments" page and disable the v2 of the quote block.
- Load the editor.
- Verify that the `core/blocks` store contains the proper block attributes under `blockTypes.core/quote`:
  -  it should have the `value` and `citation` attributes
  - it should not have the `attribution` attribute

### CODE REVIEW

File: packages/block-library/src/quote/v2/index.js
Comment (youknowriad) [inline packages/block-library/src/quote/v2/index.js:60]: What would be the issue if we use the same block.json file for both versions. I mean aside having all the old attributes in the version, which doesn't bother me much specially since we need to keep them to implement the "migration effect" behavior.

Comment (oandregal) [reply packages/block-library/src/quote/v2/index.js:60]: It was the other way around I was concerned about: I didn't want to pollute the existing quote block with the v2 attributes.

Technically, it might work in this case, as we're not _modifying_ existing atts but _adding_ new ones. I can't think of anything that would break. Though I still not a fan of this approach, I rather keep things separated.

Comment (youknowriad) [reply packages/block-library/src/quote/v2/index.js:60]: Ok, let's keep them separated for now, it doesn't hurt either.

File: packages/block-library/package.json
Comment (youknowriad) [inline packages/block-library/package.json:33]: Curious about this change, how do we decide to add a file here.

**Review by youknowriad — APPROVED** [review]
I think it's probably unrelated with this PR particularly but there's weird behavior with the quote block where if I toggle "add attribution" in the toolbar, I'm not able to see the toolbar again if I select the quote block. (Probably something wrong because we're changing the block wrapper)

Also, I noticed that in 2022, both the quote and the paragraph have a left border. (I guess due to the markup change).

Anyway, I can approve this PR (the attributes change) but would be good to check the two points above separately or track them.

File: packages/block-library/package.json
Comment (oandregal) [reply packages/block-library/package.json:33]: My understanding is that we use this to signal webpack not to remove things that aren't part of the export in those files. In theory, it should remove the `addFilter` call. In practice, though, I tested without adding this and things worked as expected. Not sure if there's some webpack config missing, but wanted to play it safe and avoid the situation in which the v2 block stops working because of some webpack config changed.

**Comment by oandregal** [comment]
Yeah, I also saw some other weird things with margins. We should review styling before making v2 the default.

> I think it's probably unrelated with this PR particularly but there's weird behavior with the quote block where if I toggle "add attribution" in the toolbar, I'm not able to see the toolbar again if I select the quote block. (Probably something wrong because we're changing the block wrapper)

Did you have the focus in the attribution or within the quote (paragraph or so)? This is a confusing interaction for me as well. It's hard to see the quote toolbar if you don't have a attribution to get the focus into. I've marked that as a task at https://github.com/WordPress/gutenberg/pull/25892#issuecomment-1076686005

**Comment by oandregal** [comment]
I'm going to merge this to unblock https://github.com/WordPress/gutenberg/pull/39718

**Pull request merged**
